java/core/src/java/org/apache/orc/impl/RecordReaderImpl.java

```diff
@@ -249,7 +249,6 @@ protected RecordReaderImpl(ReaderImpl fileReader,
               .withZeroCopy(zeroCopy)
               .build());
     }
-    this.dataReader.open();
```

Comment:
At line 273, `advanceToNextRow(reader, 0L, true);` will invoke this via `readStripeFooter`.

Reply:
@dongjoon-hyun If a read operation after the open method is interrupted, will dataReader be closed automatically (such as in a `finally` block)?

Reply:
For now, this PR doesn't aim to change `close` behavior. This only removes the unnecessary opened period.

Reply:
So this is not going to resolve https://issues.apache.org/jira/browse/SPARK-23390 Right?

Reply:
For Spark thing, let's talk on Spark-side.